hudi-common/src/main/java/org/apache/hudi/common/util/collection/RocksDBDAO.java

```diff
@@ -75,9 +75,6 @@ public RocksDBDAO(String basePath, String rocksDBBasePath) {
    * Create RocksDB if not initialized.
    */
   private RocksDB getRocksDB() {
-    if (null == rocksDB) {
```

Comment:
What if getRocksDB is called repetitively ? It should not init() every time, do we plan to add a different check in init to allow for better code coverage ?

Reply:
i looked at the entire code. it seems like `init()`  gets called from the only constructor in this class. so there will never be a case where `rocksDB == null`. this code seemed redundant and was never going to get called. that's why removed it. without this code, we are just going to return the `rocksDB` instance everytime without checking. Also checked the code and there is no where the variable is being set to null

Reply:
sounds good, thanks for the explanation

Reply:
refactored the code as you requested. lmk if this can be merged. thanks